docs/source/architecture.rst

```diff
@@ -11,7 +11,41 @@ The system architecture is shown in the figure below, illustrating the core proc
 Natural Language Understanding
 ------------------------------
 
-The :py:class:`NLU <moviebot.nlu.nlu>` component converts the natural language :py:class:`UserUtterance <moviebot.core.utterance.utterance.UserUtterance>` into a :py:class:`DialogueAct <moviebot.dialogue_manager.dialogue_act>`. This process, comprising of *intent detection* and *slot filling*, is performed based on the current dialogue state.
+The :py:class:`NLU <moviebot.nlu.nlu>` component converts the natural language :py:class:`UserUtterance <moviebot.core.utterance.utterance.UserUtterance>` into a :py:class:`DialogueAct <moviebot.dialogue_manager.dialogue_act>`. This process, comprising of *intent detection* and *slot filling*, is performed based on the current dialogue state. The component offers two distinct modules: Rule-Based and Neural using JointBERT.
+
+Rule-Based NLU
+^^^^^^^^^^^^^^
+
+The rule-based NLU module utilizes a combination of keyword extraction and heuristics to determine the user's intent and extract slot-value pairs from their utterances. This approach relies on predefined rules and patterns to interpret user input.
+
+Neural NLU with JointBERT
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Neural NLU module employs JointBERT, a neural model trained specifically for intent detection and slot filling tasks. JointBERT predicts both the intent of the user's utterance and the corresponding slot-value pairs using a trained model.
```

Comment:
The second sentence is redundant. Can be simplified:
"The Neural NLU module employs JointBERT, a neural model trained for predicting both the intent of the user's utterance and the corresponding slot-value pairs."

Reply:
Done